src/api/schema/sort.rs

```diff
@@ -0,0 +1,38 @@
+use crate::api::schema::components::ComponentsSortFieldName;
+use async_graphql::{Enum, InputObject, InputType};
+use std::cmp::Ordering;
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq)]
+pub enum Direction {
+    Asc,
+    Desc,
+}
+
+#[derive(InputObject)]
+#[graphql(concrete(name = "ComponentsSortField", params(ComponentsSortFieldName)))]
+pub struct SortField<T: InputType> {
+    pub field: T,
+    pub direction: Direction,
```

Comment:
We could optionally specify `ASC` as the default here, to avoid a few user keystrokes in the common case. Open to comments.

Reply:
Thanks, added in 98a2e49cf8813e17eeb762df87c86615b5cd1170.